src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1132,13 +1132,14 @@ internal virtual void IntersectWithEnumerable(IEnumerable<T> other)
                 if (Contains(item))
                 {
                     toSave.Add(item);
+                    DoRemove(item);
```

Comment:
Removed it. It's not really needed as `AddIfNotPresent` does the job so in the worst case we'll grow List<T> unnecessary.